connector/src/main/java/org/geysermc/connector/utils/AdvancementsUtils.java

```diff
@@ -0,0 +1,163 @@
+                                 content.append("     " + MessageTranslator.convertMessage(value.getValue().getDisplayData().getDescription().toString(), language)+ "\n\n" );
+                            }
+                    }
+                    break;
+                case 2:
+                    title = LanguageUtils.getPlayerLocaleString("advancements.end.root.title", language);
+
+                        for (Map.Entry<String, Advancement> value : session.getStoredAdvancements().entrySet()) {
+                            if (value.getKey().contains("end")){
+                                content.append(" - " + MessageTranslator.convertMessage(value.getValue().getDisplayData().getTitle().toString(), language) + "\n");
+                                content.append("     " + MessageTranslator.convertMessage(value.getValue().getDisplayData().getDescription().toString(), language)+ "\n\n" );}
+                        }
+
+                    break;
+                case 3:
+                    title = LanguageUtils.getPlayerLocaleString("advancements.adventure.root.title", language);
+
+                        for (Map.Entry<String, Advancement> value : session.getStoredAdvancements().entrySet()) {
+                            if (value.getKey().contains("adventure")){
+                                content.append(" - " + MessageTranslator.convertMessage(value.getValue().getDisplayData().getTitle().toString(), language) + "\n");
+                                content.append("     " + MessageTranslator.convertMessage(value.getValue().getDisplayData().getDescription().toString(), language)+ "\n\n" );}
+                        }
+
+                    break;
+                case 4:
+                    title = LanguageUtils.getPlayerLocaleString("advancements.husbandry.root.title", language);
+
+                        for (Map.Entry<String, Advancement> value : session.getStoredAdvancements().entrySet()) {
+                            if (value.getKey().contains("husbandry")){
+                                content.append(" - " + MessageTranslator.convertMessage(value.getValue().getDisplayData().getTitle().toString(), language) + "\n");
+                                content.append("     " + MessageTranslator.convertMessage(value.getValue().getDisplayData().getDescription().toString(), language)+ "\n\n" );}
+                        }
+
+                    break;
+
+        }
+
+
+            // Showed advancements you have earned based on category selected
+            SimpleFormWindow window = new SimpleFormWindow(title, content.toString());
+            window.getButtons().add(new FormButton(LocaleUtils.getLocaleString("gui.back", language)));
+            session.sendForm(window, ADVANCEMENTS_LIST_FORM_ID);
+        }
+
+        return true;
+    } // Handles "back" button
+    public static boolean handleListForm(GeyserSession session, String response) {
```

Comment:
There should be an empty line in between these and please use a full JavaDoc comment.

Reply:
```suggestion
    }
    
    /**
     * Handle the list form response
     *
     * @param session The session that sent the response
     * @param response The response string to parse
     * @return True if the form was parsed correctly, false if not
     */
    public static boolean handleListForm(GeyserSession session, String response) {
```